### ISSUE TITLE
how to understand the output of reg branch: 4:5 means z?

### ISSUE THREAD

**Issue by gaosanyuan** [megvii-research/PETR#174]
<img width="1353" height="933" alt="Image" src="https://github.com/user-attachments/assets/932438a4-d694-4db1-a2fe-8f1477c92aac" />

In my opinion, the output the reg branch are decided by (2, 1, 3, 2, 2),  # xy, z, size, rot, velo. Why here 4:5 means z?
Thanks

**Comment by FengLeey** [comment]
you could see this code in loss_by_feat_single function
```python
# regression L1 loss
bbox_preds = bbox_preds.reshape(-1, bbox_preds.size(-1))
normalized_bbox_targets = normalize_bbox(bbox_targets, self.pc_range)
isnotnan = torch.isfinite(normalized_bbox_targets).all(dim=-1)
bbox_weights = bbox_weights * self.code_weights

loss_bbox = self.loss_bbox(
    bbox_preds[isnotnan, :10],
    normalized_bbox_targets[isnotnan, :10],
    bbox_weights[isnotnan, :10],
    avg_factor=num_total_pos)
```
and the normalize_bbox change the GT order
```python
def normalize_bbox(bboxes, pc_range):

    cx = bboxes[..., 0:1]
    cy = bboxes[..., 1:2]
    cz = bboxes[..., 2:3]
    length = bboxes[..., 3:4].log()
    width = bboxes[..., 4:5].log()
    height = bboxes[..., 5:6].log()

    rot = -bboxes[..., 6:7] - np.pi / 2
    rot = limit_period(rot, period=np.pi * 2)
    if bboxes.size(-1) > 7:
        vx = bboxes[..., 7:8]
        vy = bboxes[..., 8:9]
        normalized_bboxes = torch.cat(
            (cx, cy, length, width, cz, height, rot.sin(), rot.cos(), vx, vy),
            dim=-1)
    else:
        normalized_bboxes = torch.cat(
            (cx, cy, length, width, cz, height, rot.sin(), rot.cos()), dim=-1)
    return normalized_bboxes
```
